### PR TITLE
spectest: support assert_unlinkable and assert_uninstantiable

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: test/spectests/spectests.cpp [hunk 1]
```diff
@@ -256,6 +256,49 @@ class test_runner
...
+                        std::move(imports->memories), std::move(imports->globals));
+                }
+                catch (const fizzy::parser_error& ex)
+                {
+                    fail(std::string{"Parsing failed with error: "} + ex.what());
+                    continue;
+                }
+                catch (const fizzy::instantiate_error&)
+                {
+                    pass();
```
Comment (axic) [inline test/spectests/spectests.cpp]: Should it check here for `instantiate` message ?

Comment (gumb0) [reply test/spectests/spectests.cpp]: Yes we can check.

File: test/spectests/spectests.cpp [hunk 2]
```diff
@@ -256,6 +257,55 @@ class test_runner
...
+                        std::move(imports.globals));
+                }
+                catch (const fizzy::parser_error& ex)
+                {
+                    fail(std::string{"Parsing failed with error: "} + ex.what());
+                    continue;
+                }
+                catch (const fizzy::instantiate_error& ex)
+                {
+                    if (ex.what() == std::string{"Start function failed to execute"})
```
Comment (axic) [inline test/spectests/spectests.cpp]: Need to also check for `type == "assert_uninstantiable"`.

File: test/spectests/spectests.cpp [hunk 3]
```diff
@@ -256,6 +257,55 @@ class test_runner
...
+                        pass();
+                    else
+                        fail(std::string{"Instantiation failed with error: "} + ex.what());
```
Comment (axic) [inline test/spectests/spectests.cpp]: Shouldn't this be pass if `type == "assert_unlinkable"`?

**Comment by gumb0** [comment]
There're some UBSan warnings in spectest run, not sure if they're caused by this PR

**Comment by axic** [comment]
>There're some UBSan warnings in spectest run, not sure if they're caused by this PR

Those been there before this PR. We need to fix them.

**Comment by axic** [comment]
I'm merging this, we fix it later too.

**Pull request merged**